libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1164,8 +1183,113 @@ std::map<std::string, Json::Value> StandardCompiler::parseAstFromInput(StringMap
 	return sourceJsons;
 }
 
+Json::Value StandardCompiler::importEVMAssembly(StandardCompiler::InputsAndSettings _inputsAndSettings)
+{
+	solAssert(_inputsAndSettings.language == "EVMAssembly");
+	solAssert(_inputsAndSettings.sources.empty());
+	solAssert(_inputsAndSettings.jsonSources.size() == 1);
+
+	if (!isBinaryRequested(_inputsAndSettings.outputSelection))
+		return Json::objectValue;
+
+	evmasm::EVMAssemblyStack stack(_inputsAndSettings.evmVersion);
+	std::string const& sourceName = _inputsAndSettings.jsonSources.begin()->first; // result of structured binding can only be used within lambda from C++20 on.
+	Json::Value const& sourceJson = _inputsAndSettings.jsonSources.begin()->second;
+	try
+	{
+		stack.analyze(sourceName, sourceJson);
+		stack.assemble();
+	}
+	catch (evmasm::AssemblyImportException const& e)
+	{
+		return formatFatalError(Error::Type::Exception, "Assembly import error: " + std::string(e.what()));
+	}
+	catch (util::Exception const& e)
```

Comment:
Wasn't this `InvalidOpcode` or something? We need handlers for exceptions that can actually happen (i.e. are errors, not assertion failures or bugs) and one catch-all for unexpected exceptions (bugs).

```suggestion
	catch (InvalidOpcode const& e)
```

Reply:
ah yeah!